runtime/parser2/lexer/tokentype.go

```diff
@@ -80,90 +84,98 @@ func (t TokenType) String() string {
 		return "EOF"
 	case TokenSpace:
 		return "space"
-	case TokenNumber:
-		return "number"
+	case TokenBinaryLiteral:
+		return "binary integer"
+	case TokenOctalLiteral:
+		return "octal integer"
+	case TokenDecimalLiteral:
+		return "decimal integer"
+	case TokenHexadecimalLiteral:
+		return "hexadecimal integer"
+	case TokenFixedPointLiteral:
+		return "fixed-point number"
 	case TokenIdentifier:
 		return "identifier"
 	case TokenString:
 		return "string"
 	case TokenPlus:
-		return "+"
+		return `'+'`
```

Comment:
Just curious why you quoted these characters in this way?

Reply:
The result of this function here is mostly a representation of the token, so I quoted all the ones that are literals (e.g `+`, `"`, etc.), but left the "set" / descriptive ones (e.g. space, comment), so they are properly rendered in error messages, e.g. "expected '+'" but "expected space"